src/latexify/codegen/function_codegen.py

```diff
@@ -536,6 +536,30 @@ def visit_IfExp(self, node: ast.IfExp) -> str:
         latex += self.visit(node)
         return latex + r", & \mathrm{otherwise} \end{array} \right."
 
+    def visit_Match(self, node: ast.Match) -> str:
+        """Visit a match node"""
+        latex = r"\left\{ \begin{array}{ll} "
+        subject_latex = self.visit(node.subject)        
+        for match_case in node.cases:
+            true_latex, cond_latex = self.visit(match_case)
+            if cond_latex:
+                latex += true_latex + r", & \mathrm{if} \ " + subject_latex + cond_latex + r" \\ "
+            else:
+                latex += true_latex + r", & \mathrm{otherwise}"
+        latex += r"\end{array} \right."
+        return latex
+    
+    def visit_match_case(self, node: ast.match_case) -> str:
```

Comment:
Wrong type: this function returns `tuple[str, str]`. If you need to implement a method that returns other types than `str`, please implement non-visitor function (functions that don't have `visit_` prefix).

I also think this visitor is not necessary. Since the `cases` member is always `match_case` and `Match` is the only syntax where `match_case` is used, it could be implemented directly in `visit_Match`.